include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -256,6 +253,45 @@ struct aws_s3_meta_request {
         struct aws_array_list event_delivery_array;
     } io_threaded_data;
 
+    /**
+     * Data for async-writes.
+     * Currently, we only allow 1 async-write at a time.
+     *
+     * When the user calls write(), they may not provide enough data for us to
+     * send an UploadPart. In that case, we immediately mark the write complete,
```

Comment:
do we do createMPU right away or wait for at least a block full of data?

Reply:
possible optimization for future is to delay until we get full part or eof and then decide on whether to do default or mpu

Reply:
Currently, we're always doing CreateMultipartUpload. Yeah, agree that's a future optimization we could do for this (or any PUT where we don't know Content-Length)